cmd/gpu_plugin/gpu_plugin.go

```diff
@@ -238,7 +349,7 @@ func main() {
 		os.Exit(1)
 	}
 
```

Comment:
I have a supplement here, if we
> add a validating webhook check about this
Then, we also need to add a check here.

Reply:
I have a concern that if we really need such a check, because `preferredAllocationPolicy` is not totally decided by `sharedDevNum`. We suggest to set `preferredAllocationPolicy` only when `sharedDevNum >1`, but if user do not set `sharedDevNum >1`, `preferredAllocationPolicy` can also work. In this situation, the three policies have no difference.

Reply:
After consideration, I think there is no need to add extra check here.